packages/block-editor/src/store/selectors.js

```diff
@@ -1243,7 +1244,11 @@ const canInsertBlockTypeUnmemoized = (
 		return defaultResult;
 	};
 
-	const blockType = getBlockType( blockName );
+	let blockType = getBlockType( blockName );
```

Comment:
You can make the check before calling `getBlockType` as there is no need to, if an `object` is passed.
I also think you should write just a small comment to explain that the case of `object` is from block directory.